### PR TITLE
Add keyboard buff75

### PR DESCRIPTION
**Pull request by Vem-596**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/herevoland/buff75/halconf.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#define HAL_USE_PWM TRUE
```
Comment (zvecr) [inline keyboards/herevoland/buff75/halconf.h:18]: Configuring PWM and then not using it for anything is incorrect.

If the intention is to configure the PWM WS2812 driver, then additional config is required.

File: keyboards/herevoland/buff75/info.json [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    "ws2812": {
+        "pin": "B2"
+    },
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "cols": ["B9", "B8", "B7", "B6", "B5", "B4", "B3", "A15", "A8", "B15", "B14", "B13", "B12", "A3", "A4" ],
+        "rows": ["A5", "B10", "B1", "B0", "A7", "A6"]
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/herevoland/buff75/info.json:47]: Providing a single layout macro, when your matrix_diagram shows multiple is not correct.

File: keyboards/herevoland/buff75/readme.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Buff75
+
+![Buff75](https://imgur.com/UjwWjYY.jpg)
+An 75%/TenKeyLess sized keyboard with 1 RGB.
+This PCB has two versions: hotswap and solder.
+
+* Keyboard Maintainer: [HereVoLand](https://github.com/Vem-596)
+* Hardware Supported: buff75
+* Hardware Availability: [HereVoLand](https://github.com/Vem-596)
```
Comment (zvecr) [inline keyboards/herevoland/buff75/readme.md:9]: This isnt a valid link to hardware

File: keyboards/herevoland/buff75/matrix_diagram.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Matrix Diagram for Buff75
```
Comment (zvecr) [inline keyboards/herevoland/buff75/matrix_diagram.md:1]: > This PCB has two versions: hotswap and solder.

If only some of the possible layout can apply to either, then the expectation is that the keyboard has 2 sub-revisions, 
```
keyboards/herevoland/buff75/soldered/keyboard.json
keyboards/herevoland/buff75/hotswap/keyboard.json
```

File: keyboards/herevoland/buff75/rules.mk
Comment (zvecr) [inline keyboards/herevoland/buff75/rules.mk:1]: This file should be removed.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/herevoland/buff75/info.json [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+                        {"label":"Ctrl", "x":0, "y":5.25, "w":1.25}, 
+                        {"label":"Win", "x":1.25, "y":5.25, "w":1.25}, 
+                        {"label":"Alt", "x":2.5, "y":5.25, "w":1.25},  
+                        {"label":"Space", "x":3.75, "y":5.25, "w":6.25},        
+                        {"label":"Alt", "x":10, "y":5.25, "w":1.25}, 
+                        {"label":"Fn", "x":11.25, "y":5.25, "w":1.25},       
+                        {"label":"Left", "x":13, "y":5.25},    
+                        {"label":"Down", "x":14, "y":5.25},         
+                        {"label":"Right", "x":15, "y":5.25}
```
Comment (zvecr) [inline keyboards/herevoland/buff75/info.json:135]: These are missing matrix locations.

**Review by Vem-596 — COMMENTED** [review]
Some changes have been made, please help me review them, thank you.

File: keyboards/herevoland/buff75/matrix_diagram.md (hunk 4 above)
Comment (zvecr) [reply keyboards/herevoland/buff75/matrix_diagram.md]: Just deleting the `matrix_diagram.md` file does not exclude you from making these changes.

Your product photo clearly shows multiple available layouts.

File: keyboards/herevoland/buff75/info.json (hunk 2 above)
Comment (zvecr) [reply keyboards/herevoland/buff75/info.json]: Just deleting the `matrix_diagram.md` file does not exclude you from making these changes.

Your product photo clearly shows multiple available layouts.

Comment (Vem-596) [reply keyboards/herevoland/buff75/info.json]: Yes, we considered canceling the solder version, but we just forgot to replace the product photo

Comment (zvecr) [reply keyboards/herevoland/buff75/info.json]: So if a PR to the-via/keyboards gets added with multiple layout support, you agree we can remove this code from the repo due to it being raised under false pretences?

Comment (Vem-596) [reply keyboards/herevoland/buff75/info.json]: Agreed, this is reasonable.

**Comment by drashna** [comment]
As of August 26, 2024, `qmk/qmk_firmware` is no longer accepting VIA-enabled keymaps as these have now transitioned to a repository under the VIA team's control.

As you've submitted a PR containing `via` or VIA-enabled keymap(s), this is your notice that they should be removed from this PR. You should now submit a secondary PR to the [VIA QMK Userspace](https://github.com/the-via/qmk_userspace_via) repository with your associated `via` or VIA-enabled keymaps instead.
Copy

**Review by drashna — APPROVED** [review]

File: keyboards/herevoland/buff75/matrix_diagram.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+            │10 │11 │12 │13 │14 │15 │16 │17 │18 │19 │1A │1B │1C │1D │3C │1E │   
+            ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┴┬──┼───┤    
+            │20   │21 │22 │23 │24 │25 │26 │27 │28 │29 │2A │2B │2C    │2D│2E │         
+            ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬─────┴──┼───┤   
+            │30    │31 │32 │33 │34 │35 │36 │37 │38 │39 │3A │3B │3D      │3E │  
+            ├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────┬───┼───┤   
+            │40      │42 │43 │44 │45 │46 │47 │48 │49 │4A │4B │4C    │4D │4E │    
+            ├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴──┬┴───┴┬─┬───┼───┼───┤ 
+            │50  │51  │52  │56                      │5A │ 5B  │ │5C │5D |5E |             
+            └────┴────┴────┴────────────────────────┴───┴─────┘ └───┴───┘───┘
```
Comment (fauxpark) [inline keyboards/herevoland/buff75/matrix_diagram.md:17]:
```suggestion
            ┌───┐ ┌───┬───┬───┬───┐ ┌───┬───┬───┬───┐ ┌───┬───┬───┬───┐ ┌───┐
            │00 │ │01 │02 │03 │04 │ │06 │07 │08 │09 │ │0A │0B │0C │0D │ │0E │
            └───┘ └───┴───┴───┴───┘ └───┴───┴───┴───┘ └───┴───┴───┴───┘ └───┘
            ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┐
            │10 │11 │12 │13 │14 │15 │16 │17 │18 │19 │1A │1B │1C │1D │3C │1E │
            ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┴┬──┼───┤
            │20   │21 │22 │23 │24 │25 │26 │27 │28 │29 │2A │2B │2C    │2D│2E |
            ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬─────┴──┼───┤
            │30    │31 │32 │33 │34 │35 │36 │37 │38 │39 │3A │3B │3D      │3E │
            ├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────┬───┼───┤
            │40      │42 │43 │44 │45 │46 │47 │48 │49 │4A │4B │4C    │4D │4E │
            ├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴──┬┴───┴┬─┬───┼───┼───┤
            │50  │51  │52  │56                      │5A │ 5B  │ │5C │5D |5E |
            └────┴────┴────┴────────────────────────┴───┴─────┘ └───┴───┘───┘
```

File: keyboards/herevoland/buff75/readme.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+An 75%/TenKeyLess sized keyboard with 1 RGB.
+
+* Keyboard Maintainer: [HereVoLand](https://github.com/Vem-596)
+* Hardware Supported: buff75
+* Hardware Availability: [BUFF Customer-Keyboard](https://shop107132374.taobao.com)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make herevoland/buff75:default
+
```
Comment (fauxpark) [inline keyboards/herevoland/buff75/readme.md:13]:
```suggestion

```

File: keyboards/herevoland/buff75/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,  KC_F2, KC_F3, KC_F4,        KC_F5, KC_F6, KC_F7,  KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_HOME,  
+        KC_GRV,  KC_1,   KC_2,  KC_3,  KC_4,  KC_5,  KC_6,  KC_7,  KC_8,    KC_9,   KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_PGUP,  
+        KC_TAB,  KC_Q,   KC_W,  KC_E,  KC_R,  KC_T,  KC_Y,  KC_U,  KC_I,    KC_O,   KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGDN,  
+        KC_CAPS, KC_A,   KC_S,  KC_D,  KC_F,  KC_G,  KC_H,  KC_J,  KC_K,    KC_L,   KC_SCLN, KC_QUOT, KC_TRNS, KC_ENT,  KC_DEL,   
+        KC_LSFT,         KC_Z,  KC_X,  KC_C,  KC_V,  KC_B,  KC_N,  KC_M,    KC_COMM,KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_END,   
+        KC_LCTL, KC_LGUI,KC_LALT,                    KC_SPC,                        KC_RALT, MO(1),   KC_LEFT, KC_DOWN, KC_RIGHT
```
Comment (fauxpark) [inline keyboards/herevoland/buff75/keymaps/default/keymap.c:26]:
```suggestion
        KC_ESC,  KC_F1,  KC_F2, KC_F3, KC_F4,        KC_F5, KC_F6, KC_F7,  KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_HOME,
        KC_GRV,  KC_1,   KC_2,  KC_3,  KC_4,  KC_5,  KC_6,  KC_7,  KC_8,    KC_9,   KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_PGUP,
        KC_TAB,  KC_Q,   KC_W,  KC_E,  KC_R,  KC_T,  KC_Y,  KC_U,  KC_I,    KC_O,   KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGDN,
        KC_CAPS, KC_A,   KC_S,  KC_D,  KC_F,  KC_G,  KC_H,  KC_J,  KC_K,    KC_L,   KC_SCLN, KC_QUOT, KC_TRNS, KC_ENT,  KC_DEL,
        KC_LSFT,         KC_Z,  KC_X,  KC_C,  KC_V,  KC_B,  KC_N,  KC_M,    KC_COMM,KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_END,
        KC_LCTL, KC_LGUI,KC_LALT,                    KC_SPC,                        KC_RALT, MO(1),   KC_LEFT, KC_DOWN, KC_RIGHT
```

File: keyboards/herevoland/buff75/keyboard.json
Comment (fauxpark) [inline keyboards/herevoland/buff75/keyboard.json:1]: Please run `qmk format-json -i` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/herevoland/buff75/readme.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+![Buff75](https://imgur.com/pQfzIZr.jpg)
+An 75%/TenKeyLess sized keyboard with 1 RGB.
+
+* Keyboard Maintainer: [HereVoLand](https://github.com/Vem-596)
+* Hardware Supported: buff75
+* Hardware Availability: [BUFF Customer-Keyboard](https://shop107132374.taobao.com)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make herevoland/buff75:default
```
Comment (zvecr) [inline keyboards/herevoland/buff75/readme.md:12]: Previous suggestion here would have been to remove trailing whitespace not the newline
```suggestion
    make herevoland/buff75:default

```

File: keyboards/herevoland/buff75/matrix_diagram.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
+# Matrix Diagram for Buff75
+
+```
+            ┌───┐ ┌───┬───┬───┬───┐ ┌───┬───┬───┬───┐ ┌───┬───┬───┬───┐ ┌───┐
+            │00 │ │01 │02 │03 │04 │ │06 │07 │08 │09 │ │0A │0B │0C │0D │ │0E │
+            └───┘ └───┴───┴───┴───┘ └───┴───┴───┴───┘ └───┴───┴───┴───┘ └───┘
+            ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┐
```
Comment (zvecr) [inline keyboards/herevoland/buff75/matrix_diagram.md:7]: Your product image, this matrix diagram, keymaps, and keyboard.json still do not line up.

Image shows a 2u backspace, while this suggests split backspace.

Comparing the 2nd row
* image.jpg - 15 keys
* keymap.c 2nd row - 15 keys
* matrix_diagram.md - 16 keys
* keyboard.json - 16 keys

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/herevoland/buff75/readme.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
```
Comment (zvecr) [inline keyboards/herevoland/buff75/readme.md:12]: Still wrong, 3rd time this is being suggested.
```suggestion
    make herevoland/buff75:default

```

File: keyboards/herevoland/buff75/keyboard.json [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+        "sleep": true,
+        "animations": {
+            "breathing": true,
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "snake": true,
+            "knight": true,
+            "christmas": true,
+            "static_gradient": true,
+            "rgb_test": true,
```
Comment (fauxpark) [inline keyboards/herevoland/buff75/keyboard.json:33]:
```suggestion
```

File: keyboards/herevoland/buff75/mcuconf.h [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include_next <mcuconf.h>
+#undef STM32_PWM_USE_TIM2
+#define STM32_PWM_USE_TIM2 TRUE
```
Comment (fauxpark) [inline keyboards/herevoland/buff75/mcuconf.h:21]: What is this for? I don't see anything that uses PWM enabled.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/herevoland/buff75/mcuconf.h
Comment (fauxpark) [inline keyboards/herevoland/buff75/mcuconf.h:1]: Please delete this file entirely.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**